Tasks/AzureRmWebAppDeployment/AzureRmWebAppDeployment.ts

```diff
@@ -37,6 +39,20 @@ async function run() {
         if (!tl.exist(webDeployPkg)) {
             throw new Error(tl.loc('Invalidwebapppackageorfolderpathprovided', webDeployPkg));
         }
+
+        var isParamFilePresentInPackage = false;
+        var isFolderBasedDeployment = !msDeployUtility.fileExists(webDeployPkg);
+        
+        if (!isFolderBasedDeployment) {
+            isParamFilePresentInPackage = await msDeployUtility.containsParamFile(webDeployPkg);
+        }
+
+        if(!tl.filePathSupplied('SetParametersFile')) {
+            setParametersFile = null;
+        }
+        else if (!msDeployUtility.fileExists(setParametersFile)) {
```

Comment:
This check should be done only if webdeploy is enabled. Should not be done in default deployment based on Kudu REST API.